src/cargo/util/toml/embedded.rs

```diff
@@ -140,8 +147,14 @@ fn expand_manifest_(script: &RawScript, config: &Config) -> CargoResult<toml::Ta
         .file_stem()
         .ok_or_else(|| anyhow::format_err!("no file name"))?
         .to_string_lossy();
-    let separator = '_';
-    let name = sanitize_package_name(file_name.as_ref(), separator);
+    let separator = if file_name.contains('_') {
+        '_'
+    } else {
+        // Since embedded manifests only support `[[bin]]`s, prefer arrow-case as that is the
+        // more common convention for CLIs
+        '-'
+    };
+    let name = sanitize_name(file_name.as_ref(), separator);
```

Comment:
Should we move these lines into `sanitize_name()`?

Reply:
Yeah, I don't think we will need it separate in the future.

It was originally out because this implemented before #12269 and we needed the separator for the bin name.  I am wanting to avoid doing that in the future and so the only place I can see us doing this is in non-user-visible locations where it won't matter what we do (target dir), so I don't think we'll need it again.